sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1774,6 +1773,19 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  object IntervalStyle extends Enumeration {
+    val SQL_STANDARD, MULTI_UNITS = Value
+  }
+
+  val INTERVAL_STYLE = buildConf("spark.sql.IntervalOutputStyle")
+    .doc("Display format for interval values. The value SQL_STANDARD will produce output" +
+      " matching SQL standard interval literals. The value MULTI_UNITS (which is the default)" +
+      " will produce output in form of value unit pairs, i.e. '3 year 2 months 10 days'")
+    .stringConf
+    .transform(_.toUpperCase(Locale.ROOT))
+    .checkValues(IntervalStyle.values.map(_.toString))
+    .createWithDefault(IntervalStyle.MULTI_UNITS.toString)
```

Comment:
I personally think `ansiEnabled` is enough for this feature. Any concern?

Reply:
yes, I guess some users may already rely on the output string